src/components/DataControllerMixin/DataControllerMixin.js

```diff
@@ -86,62 +109,84 @@ var ModalMixin = {
         variable: props.variable_id,
         area: props.area || "",
       }
-    }).then(this.validateTimeseriesData);
+    }).then(validateAnnualCycleData);
   },
 
-  validateStatsData: function (response) {
-    if(_.isEmpty(response.data) || (typeof response.data == "string")) {
-      throw new Error("Error: statistical data unavailable for this model");
-    }
-    for(var file in response.data) {
-      if(_.some('mean stdev min max median ncells'.split(' '),
-          attr => !(attr in response.data[file]) || isNaN(response.data[file][attr])) ||
-          _.some('units time'.split(' '),
-              attr => !(attr in response.data[file]))) {
-        throw new Error("Error: statistical data for this model is incomplete");
-      }
-    }
-    return response;
-    },
-
-    validateTimeSeriesData: function(response) {
-      if(_.isEmpty(response.data)) {
-        throw new Error("Error: timeseries data is unavailable for this model.");
-      }
-      if(!_.every('id units data'.split(' '), attr => attr in response.data)) {
-        throw new Error("Error: timeseries data for this model is incomplete");
-      }
-      return response;
-    },
-
-    validateAnnualData : function(response){
-      if(_.isEmpty(response.data)) {
-        throw new Error("Error: annual data unavailable for this model.");
-      }
-      for(var run in response.data) {
-        if(!('data' in response.data[run]) || !('units' in response.data[run])) {
-          throw new Error("Error: annual data for this model is incomplete.");
-        }
-      }
-      return response;
-    },
-
+    /*
+     * this function is called to display any error generated in the 
+     * process of showing a graph or table, so it handles networking
+     * errors thrown by axios calls and errors thrown by validators 
+     * and parsers, which have different formats.
+     */
     displayError: function(error, displayMethod) {
-      if(error.response) { // data server sent a non-200 response
+      if(error.response) { // axios error: data server sent a non-200 response
         displayMethod("Error: " + error.response.status + " received from data server.");
-      }else if(error.request) { // data server didn't respond
+      }else if(error.request) { // axios error: data server didn't respond
         displayMethod("Error: no response received from data server.");
-      }else {  // either a failed data validation
-        // or a generic and somewhat unhelpful "Network Error" from axios
+      }else {  
+        // either an error thrown by a data validation function,
+        // an error thrown by the DataGraph or DataTable parsers,
+        // or the generic and somewhat unhelpful "Network Error" from axios
```

Comment:
Usually axios provides an HTTP status code in `error.response`. Once in a while it doesn't, and provides only the "Network Error" message without populating `error.response` or `error.request`. [We are not the only ones who have encountered intermittent lack of http status on axios errors.](https://github.com/mzabriskie/axios/issues/383)

As far as I could tell from testing, the lack-of-http-status-code in our case happened when the backend's status was 500, but not every time. Perhaps it was related to how long the backend took to return a response.

This may be fixed, I see axios' most recent update [changelog](https://github.com/mzabriskie/axios/blob/master/CHANGELOG.md) (0.16.2) includes
- Including underlying request in errors